### PR TITLE
Reply to comment functionality

### PR DESCRIPTION
**Pull request by gdpelican**
Adds in-reply-to functionality to angular comments

To wit:

**Replying to a comment:**
screenshot from 2015-02-02 17 43 30

**Viewing a comment reply:**
screenshot from 2015-02-02 17 43 48

**Viewing a comment reply notification:**
screenshot from 2015-02-02 17 44 27

### CODE REVIEW

**Comment by rdbartlett** [comment]
looks beautiful BJK

On 2 February 2015 at 17:53, James Kiesel <notifications@github.com> wrote:

> Adds in-reply-to functionality to angular comments
>
> To wit:
>
> *Replying to a comment:*
> [image: screenshot from 2015-02-02 17 43 30]
> <https://cloud.githubusercontent.com/assets/750477/5995967/f36fb98a-ab03-11e4-8531-ee73160f39e8.png>
>
> *Viewing a comment reply:*
> [image: screenshot from 2015-02-02 17 43 48]
> <https://cloud.githubusercontent.com/assets/750477/5995968/f998ca0e-ab03-11e4-9370-b72d386590e1.png>
>
> *Viewing a comment reply notification:*
> [image: screenshot from 2015-02-02 17 44 27]
> <https://cloud.githubusercontent.com/assets/750477/5995972/008ee9ec-ab04-11e4-9d04-b26fd72c5fb4.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/loomio/loomio/pull/1863
> Commit Summary
>
>    - connect in-reply-to at-mention feature
>    - Don't apply at-mention to body field
>    - Merge branch 'in-reply-to' of https://github.com/loomio/loomio into
>    in-reply-to
>    - Add notifications for comment replies
>    - Merge branch 'angular' into angular-in-reply-to
>    - Merge branch 'master' into angular-in-reply-to
>
> File Changes
>
>    - *M* app/models/event.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-0> (2)
>    - *A* app/models/events/comment_replied_to.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-1> (26)
>    - *M* app/models/events/new_comment.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-2> (2)
>    - *M* app/serializers/event_serializer.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-3> (4)
>    - *M* config/locales/client.en.yml
>    <https://github.com/loomio/loomio/pull/1863/files#diff-4> (2)
>    - *M* config/locales/en.yml
>    <https://github.com/loomio/loomio/pull/1863/files#diff-5> (1)
>    - *M* extras/notification_item.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-6> (2)
>    - *A* extras/notification_items/comment_replied_to.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-7> (18)
>    - *M* lineman/app/models/comment_model.coffee
>    <https://github.com/loomio/loomio/pull/1863/files#diff-8> (3)
>    - *M* lineman/app/models/user_model.coffee
>    <https://github.com/loomio/loomio/pull/1863/files#diff-9> (3)
>    - *M*
>    lineman/app/modules/discussion_page/comment_form/comment_form.haml
>    <https://github.com/loomio/loomio/pull/1863/files#diff-10> (1)
>    - *M*
>    lineman/app/modules/discussion_page/comment_form/comment_form_controller.coffee
>    <https://github.com/loomio/loomio/pull/1863/files#diff-11> (14)
>    - *M* lineman/app/modules/group_page/group_form.haml
>    <https://github.com/loomio/loomio/pull/1863/files#diff-12> (31)
>    - *M* lineman/app/modules/navbar/navbar_notifications_controller.coffee
>    <https://github.com/loomio/loomio/pull/1863/files#diff-13> (1)
>    - *A*
>    lineman/app/modules/navbar/notification_items/comment_replied_to.haml
>    <https://github.com/loomio/loomio/pull/1863/files#diff-14> (7)
>    - *A* spec/extras/notification_items/comment_replied_to_spec.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-15> (33)
>    - *A* spec/models/events/comment_replied_to_spec.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-16> (26)
>    - *A* spec/models/events/new_comment_spec.rb
>    <https://github.com/loomio/loomio/pull/1863/files#diff-17> (32)
>
> Patch Links:
>
>    - https://github.com/loomio/loomio/pull/1863.patch
>    - https://github.com/loomio/loomio/pull/1863.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/1863>.
>



-- 

Richard D. Bartlett
Loomio co-founder
rich@loomio.org

<http://loomio.org/>

File: app/models/events/new_comment.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@ def self.publish!(comment)
     DiscussionReader.for(user: comment.author,
                          discussion: comment.discussion).follow!
 
+    Events::CommentRepliedTo.publish! comment if comment.parent.present?
```
Comment (robguthrie) [inline app/models/events/new_comment.rb]: please abstract this.. comment.is_reply?

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -53,6 +54,7 @@ en:
     motion_closed_by_expiry: "Proposal has closed"
     new_vote: "%{position} on the proposal"
     new_comment: "commented on the discussion"
+    comment_replied_to: "replied to your comment in the discussion"
```
Comment (robguthrie) [inline config/locales/client.en.yml]: Please include the whole message in the translation string.. user names and discussion titles

Comment (rdbartlett) [inline config/locales/client.en.yml]: yep +1. as a rule, it's much easier to translate long strings with many
variables. wherever possible the translation string should be a whole
sentence.

On 2 February 2015 at 18:49, Robert Guthrie <notifications@github.com>
wrote:

> In config/locales/client.en.yml
> <https://github.com/loomio/loomio/pull/1863#discussion_r23907859>:
>
> > @@ -53,6 +54,7 @@ en:
> >      motion_closed_by_expiry: "Proposal has closed"
> >      new_vote: "%{position} on the proposal"
> >      new_comment: "commented on the discussion"
> > +    comment_replied_to: "replied to your comment in the discussion"
>
> Please include the whole message in the translation string.. user names
> and discussion titles
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/1863/files#r23907859>.
>



-- 

Richard D. Bartlett
Loomio co-founder
rich@loomio.org

<http://loomio.org/>

**Comment by robguthrie** [comment]
Looking good @gdpelican. 

I really want to break the bad pattern started with the notification strings, where we don't have the values in the translation. If you have problems please ask me.

Also would be really great to get a integration test for this if you have time.

**Pull request merged**